core/src/main/kotlin/behavior/MessageBehavior.kt

```diff
@@ -212,13 +235,51 @@ interface MessageBehavior : KordEntity, Strategizable {
  * @return The edited [Message].
  *
  * @throws [RestRequestException] if something went wrong during the request.
+ * @throws [IllegalStateException] if this message is a webhook message (See [editWebhookMessage]).
+ * @see editWebhookMessage
  */
 @OptIn(ExperimentalContracts::class)
 suspend inline fun MessageBehavior.edit(builder: MessageModifyBuilder.() -> Unit): Message {
     contract {
         callsInPlace(builder, InvocationKind.EXACTLY_ONCE)
     }
-    val response = kord.rest.channel.editMessage(channelId = channelId, messageId = id, builder = builder)
+
+    require(webhookId == null) { "Cannot perform edits on webhook messages use editWebhook() instead" }
+
+    val response =
+        kord.rest.channel.editMessage(channelId = channelId, messageId = id, builder = builder)
+    val data = MessageData.from(response)
+
+    return Message(data, kord)
+}
+
+/**
+ * Requests to edit this message.
+ *
+ * @return The edited [Message].
+ *
+ * @throws [RestRequestException] if something went wrong during the request.
+ * @throws [IllegalStateException] if this message is a normal message (see [edit]).
+ * @see edit
+ */
+@OptIn(ExperimentalContracts::class)
+suspend inline fun MessageBehavior.editWebhookMessage(
+    token: String,
+    builder: EditWebhookMessageBuilder.() -> Unit
+): Message {
+    contract {
+        callsInPlace(builder, InvocationKind.EXACTLY_ONCE)
+    }
+
+    require(webhookId != null) { "Cannot perform edits on  non webhook messages use edit() instead" }
```

Comment:
Why is this nullable in the first place?

Reply:
because both bot messages and webhook messasges are the same object and only one of them has a webhook ID

Reply:
this shouldn't be here then, this should be in the webhook service(???)

Reply:
nvm, I misunderstood, but you do expect a webhook id in the behavior it can't be nullable since it's a requirement to call the endpoint

Reply:
> nvm, I misunderstood, but you do expect a webhook id in the behavior it can't be nullable since it's a requirement to call the endpoint

The problem is we have no WebhookMessage or similar we only have Message and MessageBehvaior

Currently, you can call MessageBehavior.edit{} and it will give you a HTTP error, because it calls the normal message edit endpoint